overlord/snapstate/snapstate.go

```diff
@@ -720,6 +729,12 @@ func Update(st *state.State, name, channel string, revision snap.Revision, userI
 		return nil, err
 	}
 	if len(tts) == 0 && len(updates) == 0 {
```

Comment:
no, sadly as I explained there's a case now where len(updates) != 0 but we don't do a real refresh, I think this happens when new aliases are added for foo (but foo revision stays the same), see the reportUpdated code in doUpdate, it used both when doInstall and when newAutoAliases.  I think now updates is only not touched when we only remove aliases (this all a bit arbitrary sorry, that's why I said we need to bite the bullet and give a richer return to doUpdate and friends)